src/main/java/com/contentful/java/cda/AbsQuery.java

```diff
@@ -186,10 +186,11 @@ public Query select(String... selections) {
       }
     }
 
-    if (!name.startsWith("sys.") && !name.startsWith("fields.")) {
+    if ((!name.startsWith("sys.") && !name.startsWith("fields."))
+        && !CDAContentType.class.isAssignableFrom(type)) {
       throw new IllegalArgumentException("Please specify either a \"sys.\" or a \"fields.\" " +
           "attribute to be searched for. (Remember to specify a ContentType for \"fields.\" " +
-          "searches.)");
+          "searches on Entries only.)");
```

Comment:
This reads a bit odd to me. Perhaps: "Please specify either a sys or a fields attribute. A ContentType is mandatory when searching on Entry fields."